contents/dart/Dart.md

```diff
@@ -0,0 +1,228 @@
+
+### Benefit: Named parameters
+This is what makes Dart code look like JavaScript code (or JSON-like, as mentioned before). In languages like Java, the order of a function's parameters is specified by the function's definition, and if you only look at the function call, it might be difficult for you to figure out what those parameters mean. One classic example is the `put(K key, V value)` method of  `Map<K, V>` interface in Java - if you only see `m.put(1, 2)`, can you tell which one is key without prior knowledge?
+
+That's when named parameters become handy:
+
+When calling a function, you can specify named parameters using `paramName: value`. For example:
+```dart
+enableFlags(bold: true, hidden: false);
+```
+
+When defining a function, use `{param1, param2, ...}` to specify named parameters:
+```dart
+/// Sets the [bold] and [hidden] flags ...
+void enableFlags({bool bold, bool hidden}) {...}
+```
+
+Note that if named parameters are used, the order no longer matters, you can write `enableFlags(hidden: false, bold: true);` as well. This feature is especially useful for declarative UI as the properties of a widget are, most of the time, parameters passed to the constructor of the object. Below is a concrete example in Flutter. Imagine how messy and frustrating the code would look like if the parameters are not named and we have to refer to the documentation for the exact order of them!
+
+```dart
+// Copyright 2018 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Welcome to Flutter',
+      home: Scaffold(
+        appBar: AppBar(
+          title: Text('Welcome to Flutter'),
+        ),
+        body: Center(
+          child: Text('Hello World'),
+        ),
+      ),
+    );
+  }
+}
+```
+
+<panel header="DartPad example"><iframe style="width: 100%; height: 400px;" src="https://dartpad.dev/embed-flutter.html?id=719143b1a625dbdefb853c96de0ddbc2&split=50"></iframe></panel>
+
+Note that **named** parameters are a type of [**optional** parameters](https://dart.dev/guides/language/language-tour#optional-parameters) (i.e. you can omit those parameters and they will be assigned `null` or the default value upon entry of the function). You can also make use of [`@required`](https://pub.dev/documentation/meta/latest/meta/required-constant.html) annotation to make certain named parameters compulsory.
+
+### Benefit (advanced): Interoperability
```

Comment:
Might be better to move this to a resource further down the line for the reader to pursue since its an 'advanced' topic